hyperpay/views.py

```diff
@@ -248,21 +269,19 @@ def get(self, request, encrypted_resource_path=None):
         check_status = self._get_check_status(request)
 
         try:
-            status = PaymentStatus.PENDING
             if check_status:
                 verification_response, status = self._verify_status(resource_path)
-                if (verification_response and isinstance(verification_response, dict) and
-                        verification_response.get('merchantTransactionId')):
-                    transaction_id = verification_response['merchantTransactionId']
+
+            if verification_response and 'merchantMemo' in verification_response:
+                transaction_id = verification_response['merchantMemo']
+                basket_id = OrderNumberGenerator().basket_id(transaction_id)
+                basket = self._get_basket(basket_id)
 
             if status == PaymentStatus.FAILURE:
                 return redirect(reverse('payment_error'))
-            if status == PaymentStatus.PENDING:
+            elif status == PaymentStatus.PENDING:
```

Comment:
in this case it doesn't make any difference, probably pylint errors since that elif is after a return statement , but this is part of a testing that I made so I shouldn't have included